st2common/st2common/services/packs.py

```diff
@@ -92,6 +93,9 @@ def _fetch_and_compile_index(index_urls, logger=None, proxy_config=None):
         if http_proxy:
             proxies_dict["http"] = http_proxy
 
+        if no_proxy:
+            proxies_dict["no"] = no_proxy
```

Comment:
@minsis Did you test it BTW and confirm no_proxy setting works this way via `no` key?

Reply:
I swear this was working, but I guess not.

I'm pretty sure the `no` key is correct. I looked in the requests library and when building the proxy info it does a split. Hence the reason the other kyes are `http` and `https`.

Looking at the requests docs they never mention no_proxy, but its mentioned in a release note saying its supposed to be `no_proxy`.

Regardless `no` and `no_proxy` doesn't even seem to work at all. I tested this directly using the requests library.

Reply:
https://github.com/psf/requests/issues/4871

Looks like maybe this has been broken for quite some time. There's a PR to fix it, but its still open for 15 months.

Reply:
> I swear this was working, but I guess not.

Yeah, I'm just asking.

I'm good to merge it if you say if the code change was tested manually and works in your environment.
If you can confirm this is :+1: and fixes your case, - just let me know in the PR as the requests doc is a bit confusing indeed.

Reply:
Add `no` or `no_proxy` doesn't work unfortunately. So I guess back to square one.

My thought now is to maybe use the same logic the PR is using. Basically if it detects the URL being used is in `no_proxy` it will remove the proxy info from the request.